nemo/collections/nlp/modules/common/megatron/megatron_bert.py

```diff
@@ -56,17 +63,34 @@ def __init__(self, model_name, config, vocab_file):
 
         config['onnx_safe'] = True
 
+        # if 'model_parallel_size' in config:
+        if self._model_parallel_size is not None:
+            app_state = AppState()
+
+            # must be set for model parallel megatron-lm
+            os.environ["WORLD_SIZE"] = str(app_state.world_size)
+
```

Comment:
Setting world size here with the config/trainer doesn't work?

Reply:
That's right, megatron-lm uses that specific global variable which isn't set when we are instantiating megatron bert in nemo.